clang/include/clang/CIR/Dialect/IR/CIRAttrs.td

```diff
@@ -617,6 +617,140 @@ def DynamicCastInfoAttr
   }];
 }
 
+//===----------------------------------------------------------------------===//
+// AddressSpaceAttr
+//===----------------------------------------------------------------------===//
+
+// TODO: other CIR AS cases
+def AS_Target : I32EnumAttrCase<"target", 21>;
+
+def AddressSpaceAttr : CIR_Attr<"AddressSpace", "addrspace"> {
+
+  let summary = "Address space attribute for pointer types";
+  let description = [{
+    The address space attribute models `clang::LangAS` rather than the LLVM
+    address space, which means it's not yet converted by the address space map
+    to carry target-specific semantics.
+
+    The representation is one-to-one except for `LangAS::Default`, which
+    corresponds to a null attribute instead.
+  }];
+
+  let parameters = (ins "int64_t":$value);
```

Comment:
@Naghasan was pointing out that clang only uses 23 bits to encode address spaces, so this could be `int32_t`. Reference: https://github.com/llvm/clangir/blob/c28908396a3ba7bda6345907233e4f5c4e53a33e/clang/include/clang/AST/Type.h#L357-L359